src/app/zap-templates/templates/app/MatterIDL.zapt

```diff
@@ -61,40 +61,124 @@
+        {{#first}}
+          {{~new_line 1~}}{{~indent 1~}}request struct {{asUpperCamelCase parent.commandName}}Request {
+        {{/first}}    
+            {{~indent 2~}}{{> idl_structure_member}}
+        {{#last}}
 
-    {{/chip_cluster_command_arguments}}
-  }
+          {{~indent 1~}}}
+        {{/last}}
+      {{/zcl_command_arguments}}
+    {{/all_outgoing_commands_for_cluster}}
+  {{/if}}
+  {{#if (is_client side)}}
+    {{#all_incoming_commands_for_cluster name 'client'}}
+      {{#zcl_command_arguments}}
+        {{#first}}
+          {{~new_line 1~}}{{~indent 1~}}response struct {{asUpperCamelCase parent.commandName}} {
+        {{/first}}    
+            {{~indent 2~}}{{> idl_structure_member}}
+        {{#last}}
+
+          {{~indent 1~}}}
+        {{/last}}
+      {{/zcl_command_arguments}}
+    {{/all_incoming_commands_for_cluster}}
   {{/if}}
-  {{/chip_cluster_commands}}
-  {{#chip_cluster_responses}}
+  {{#if (is_server side)}}
+    {{#all_outgoing_commands_for_cluster name 'server'}}
+      {{#zcl_command_arguments}}
+        {{#first}}
+          {{~new_line 1~}}{{~indent 1~}}response struct {{asUpperCamelCase parent.commandName}} {
+        {{/first}}    
+            {{~indent 2~}}{{> idl_structure_member}}
+        {{#last}}
 
-  response struct {{asUpperCamelCase name}} {
-    {{#chip_cluster_response_arguments}}
-    {{> idl_structure_member}}
+          {{~indent 1~}}}
+        {{/last}}
+      {{/zcl_command_arguments}}
+    {{/all_outgoing_commands_for_cluster}}
+  {{/if}}
+  {{#if (is_server side)}}
+    {{#all_incoming_commands_for_cluster name 'server'}}
+      {{#first}}
 
-    {{/chip_cluster_response_arguments}}
-  }
-  {{/chip_cluster_responses}}
-  {{#chip_cluster_commands}}
-  {{#first}}
-
-  {{/first}}
-  {{#if mustUseTimedInvoke}}
-  timed command
-  {{~else}}
-  command
-  {{~/if}} {{asUpperCamelCase name}}(
-    {{~#if arguments~}}
-       {{asUpperCamelCase name}}Request
-    {{~/if~}}
-  ): {{asUpperCamelCase responseName}} = {{code}};
-  {{/chip_cluster_commands}}
+      {{/first}}
+      {{#if mustUseTimedInvoke}}
+        {{~indent 1~}}timed command
+      {{~else}}
+        {{~indent 1~}}command
+      {{~/if}} {{asUpperCamelCase commandName}}(
+      {{~#zcl_command_arguments~}}
+        {{~#first~}}
+          {{asUpperCamelCase parent.commandName}}Request
+        {{~/first~}}
+      {{~/zcl_command_arguments~}}
+      ): {{#if responseName~}}
+        {{~responseName~}}
+      {{~else~}}
+        {{~#if_command_has_response commandName ''~}}
+          {{~commandName~}}Response
+        {{~else~}}
+          {{~#if_command_has_response commandName 'Request'~}}
+            {{~replace_string commandName 'Request' ''}}Response
+          {{~else~}}
+            DefaultSuccess
+          {{~/if_command_has_response~}}
+        {{~/if_command_has_response~}}
+      {{~/if~}}
+      {{~indent 0}} = {{code}};
+    {{/all_incoming_commands_for_cluster}}
+  {{/if}}
+  {{#if (is_client side)}}
```

Comment:
This seems like a lot of duplicated code. Why not have the javascript split since these are 'all commands for the current side' ?

The templates are ridiculously complex as is...